interop-cats/jvm/src/main/scala/scalaz/zio/interop/stream/Stream.scala

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2017-2019 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package scalaz.zio.interop.stream
+
+import cats.~>
+import cats.effect.Effect
+import cats.implicits._
+import scalaz.zio.{Chunk, Queue, ZIO, ZManaged, ZSchedule}
+import scalaz.zio.stream.{Take, ZSink, ZStream}
+
+final class Stream[F[+ _], +A] private[stream] (private[stream] val underlying: ZStream[Any, Throwable, A])(
+  implicit liftIO: ZIO[Any, Throwable, ?] ~> F,
+  liftZIO: F ~> ZIO[Any, Throwable, ?],
+  effect: Effect[F]
+) { self =>
+
+  import Stream.Fold
+
+  def fold[A1 >: A, S]: Fold[F, A1, S] = ???
+
+  def foldLeft[A1 >: A, S](s: S)(f: (S, A1) => F[S]): F[S] = {
+    (fold[A, S]: F[(S, S => Boolean, (S, A) => F[S]) => F[S]]).flatMap(_(s, _ => false, f))
+  }
+
+  def ++[A1 >: A](that: => Stream[F, A1]): Stream[F, A1] =
+    new Stream(underlying ++ that.underlying)
+
+  def drain: Stream[F, Nothing] =
+    new Stream(underlying.drain)
+
+  def filter(pred: A => Boolean): Stream[F, A] =
+    new Stream(underlying.filter(pred))
+
+  def filterM(pred: A => F[Boolean]): Stream[F, A] =
+    new Stream(underlying.filterM(pred.andThen(liftZIO.apply)))
+
+  def filterNot(pred: A => Boolean): Stream[F, A] =
+    filter(a => !pred(a))
+
+  def foreachWhile(f: A => F[Boolean]): F[Unit] =
+    liftIO.apply(underlying.foreachWhile(f.andThen(liftZIO.apply)))
+
+  def collect[B](pf: PartialFunction[A, B]): Stream[F, B] =
+    new Stream(underlying.collect(pf))
+
+  def drop(n: Int): Stream[F, A] =
+    new Stream(underlying.drop(n))
+
+  def dropWhile(pred: A => Boolean): Stream[F, A] =
+    new Stream(underlying.dropWhile(pred))
+
+  def flatMap[B](f: A => Stream[F, B]): Stream[F, B] =
+    new Stream(underlying.flatMap(f.andThen(_.underlying)))
+
+  def foreach(f: A => F[Unit]): F[Unit] =
+    liftIO(underlying.foreach(f.andThen(liftZIO.apply)))
+
+  def forever: Stream[F, A] =
+    new Stream(underlying.forever)
+
+  def map[B](f: A => B): Stream[F, B] =
+    new Stream(underlying.map(f))
+
+  // TODO Should we expose Chunk to clients?
```

Comment:
Yes, for sure, since there's nothing like `Chunk` in Cats Effect or Cats or Scala stdlib (now FS2 has something like it).

Reply:
I suppose we should then add interop for Chunk as well at some point?

Reply:
Yes, although I'm not sure what we would consider equivalent to it...